app/models/concerns/fog_extensions/vsphere/server.rb

```diff
@@ -35,7 +35,14 @@ def scsi_controller_type
       end
 
       def select_nic(fog_nics, nic)
-        fog_nics.detect {|fn| fn.network == nic.compute_attributes['network']} # grab any nic on the same network
+        nic_attrs = nic.compute_attributes
+        selected_nic =   fog_nics.detect { |fn| fn.network == nic_attrs['network'] } # grab any nic on the same network
+        selected_nic ||= if service.get_network(nic_attrs['network'], datacenter).key?(:id)
```

Comment:
With the extra call, I'd suggest now assigning the result of service.get_network to a variable, so we're not calling it multiple times (vSphere's API can be very slow.)

Perhaps:

<pre>
unless selected_nic
  vm_network = service.get_network(nic_attrs['network'], datacenter)
  if vm_network && vm_network.key?(:id)
    fog_nics.detect { |fn| fn.network == vm_network[:id] }
  end
end
</pre>